### ISSUE TITLE
Mkpassword not working

### ISSUE THREAD

**Issue by lisbonjoker** [ircd-hybrid/ircd-hybrid#29]
```
mavis:~/hybrid/bin$ ./mkpasswd -6 -p morte
Segmentation fault
mavis:~/hybrid/bin$ ./mkpasswd -5 -s abcd -p morte
Segmentation fault
mavis:~/hybrid/bin$ ./mkpasswd -6 -s abcd -p morte
Segmentation fault
```
Everytime

**Comment by lisbonjoker** [comment]
```
 ./mkpasswd -6 -s 3dr -p test
Segmentation fault
```
not even the example works

**Comment by miwob** [comment]
Cannot reproduce. Please provide more details about the platform, the exact version of ircd-hybrid, and whether you built from sources yourself.

**Comment by lisbonjoker** [comment]
I was following this tutorial https://www.howtoforge.com/tutorial/irc-server-with-ircd-hybrid-and-anope-on-ubuntu-15-04/ but used the latest versions of IRCD-Hybrid and Anope

**Comment by miwob** [comment]
If not already enabled, please enable core dump generation and provide a back trace of that core.

**Comment by lisbonjoker** [comment]
I added ```#include <crypt.h>``` to https://github.com/ircd-hybrid/ircd-hybrid/blob/be8d512bcf409208e495eb36691797a940fb00e3/tools/mkpasswd.c and compiled just mkpasswd again and now it works
```gcc mkpasswd.c -lcrypt -o mkpasswd```

**Comment by miwob** [comment]
What OS/version are you running this on?

**Comment by lisbonjoker** [comment]
> What OS/version are you running this on?

Ubuntu 18.04.5 LTS x86_64

**Comment by miwob** [comment]
Were there any compiler warnings while compiling the mkpasswd.c file?

**Comment by lisbonjoker** [comment]
```mkpasswd.c: In function 'main':
mkpasswd.c:159:14: warning: implicit declaration of function 'crypt'; did you mean 'creat'? [-Wimplicit-function-declaration]
   if ((ret = crypt(plaintext, salt)))
              ^~~~~
              creat
mkpasswd.c:159:12: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
   if ((ret = crypt(plaintext, salt)))
            ^
/tmp/ccmwd16A.o: In function `main':
mkpasswd.c:(.text+0x2c0): undefined reference to `crypt'
collect2: error: ld returned 1 exit status```

**Comment by lisbonjoker** [comment]
Thus adding ```#include <crypt.h>``` it worked.

**Comment by miwob** [comment]
This should be fixed now with 8b9bb81 for older glibc versions.
